### PR TITLE
177702955 wip estimators transition

### PR DESCRIPTION
**Pull request by simonwa7**

### CODE REVIEW

File: src/python/zquantum/core/wip/estimators/estimation.py
Comment (mstechly) [inline src/python/zquantum/core/wip/estimators/estimation.py:201]: I think it's better to avoid `append` if you can use list comprehension.
See [here](https://stackoverflow.com/questions/30245397/why-is-a-list-comprehension-so-much-faster-than-appending-to-a-list).

File: src/python/zquantum/core/wip/estimators/estimation_interface.py
Comment (mstechly) [inline src/python/zquantum/core/wip/estimators/estimation_interface.py:34]: No kwargs in the end?

File: steps/wip/measure.py
Comment (mstechly) [inline steps/wip/measure.py:37]: It doesn't look to me like this function needs to be in wip. Same for many others.

File: steps/wip/measure.py
Comment (mstechly) [inline steps/wip/measure.py:189]: I've deleted this step altogether in #268 .

File: src/python/zquantum/core/wip/cost_function.py
Comment (simonwa7) [reply src/python/zquantum/core/wip/cost_function.py:121]: Agreed. It should probably be moved to `zquantum.core.utils`, but I would prefer to wait until the wip is merged into core since it's a relatively low-impact switch. Putting a chore into tracker

Comment (simonwa7) [reply src/python/zquantum/core/wip/cost_function.py:121]: https://www.pivotaltracker.com/story/show/177854476

File: src/python/zquantum/core/wip/cost_function.py
Comment (simonwa7) [reply src/python/zquantum/core/wip/cost_function.py]: Resolved here: ad6d339

File: src/python/zquantum/core/wip/cost_function.py
Comment (simonwa7) [reply src/python/zquantum/core/wip/cost_function.py]: Resolved here: 2c72c88

File: src/python/zquantum/core/wip/estimators/estimation_interface.py
Comment (simonwa7) [reply src/python/zquantum/core/wip/estimators/estimation_interface.py:34]: Nope - I thought we agreed we would use partials to avoid having kwargs

File: src/python/zquantum/core/wip/cost_function.py
Comment (simonwa7) [reply src/python/zquantum/core/wip/cost_function.py]: Resolved here: 367228a

File: src/python/zquantum/core/wip/estimators/estimation_interface.py
Comment (mstechly) [reply src/python/zquantum/core/wip/estimators/estimation_interface.py:34]: I thought we agreed on the opposite 😆 
Both are fine with me ;)

File: src/python/zquantum/core/wip/estimators/estimation.py
Comment (simonwa7) [reply src/python/zquantum/core/wip/estimators/estimation.py]: Resolved here: d4222b2

File: steps/wip/measure.py
Comment (simonwa7) [reply steps/wip/measure.py:37]: I thought it would just be easier to replace the whole file since many of the imports changed

File: steps/wip/measure.py
Comment (simonwa7) [reply steps/wip/measure.py]: Deleted here: d206a43

File: steps/wip/measure.py
Comment (mstechly) [reply steps/wip/measure.py:37]: I'd say it's riskier, as your changes affect more code.
And harder to review, cause it's not obvious what has actually changed ;)

File: src/python/zquantum/core/wip/cost_function.py
Comment (mstechly) [reply src/python/zquantum/core/wip/cost_function.py:129]: Sounds fine!

**Review by mstechly — APPROVED** [review]

File: src/python/zquantum/core/wip/estimators/estimation.py
Comment (mstechly) [inline src/python/zquantum/core/wip/estimators/estimation.py:188]: I don't like `by_averaging` name.
I think the `BasicEstimator` was a good name, so I tried to emulate that with `naively`. 
What about `basic_estimate_expectation_values`, `standard_estimate_expectation_values`or sth like that?

Comment (simonwa7) [reply src/python/zquantum/core/wip/estimators/estimation.py:188]: I respectfully disagree. 

`BasicEstimator` (and similarly `basic_estimate...` and `standard_estimate...`) doesn't really give any indication as to what is being done. It's only "basic" since averaging is what is considered commonplace in the industry. `by_averaging` on the other hand, explicitly communicates what is being done.

File: steps/wip/measure.py
Comment (simonwa7) [reply steps/wip/measure.py:37]: Not trying to fight you on this one, but genuinely curious how you would then distinguish between removing a piece of code in a wip module because it is unaffected vs it no longer being needed.

Comment (mstechly) [reply steps/wip/measure.py:37]: That's a very good point :D

**Pull request merged**
